src/modules/csm/tree.py

```diff
@@ -1,18 +1,35 @@
 import json
 from dataclasses import dataclass
-from typing import Self, Sequence
+from typing import Literal, Self, Sequence
 
 from hexbytes import HexBytes
-from oz_merkle_tree import StandardMerkleTree
+from oz_merkle_tree import Dump, StandardMerkleTree
 
 from src.modules.csm.types import RewardTreeLeaf
+from src.providers.ipfs.cid import CID
+
+type StateCID = CID | Literal[""]
```

Comment:
why `""` is legit value?

Reply:
By loading tree from dump we don't care about this value. And we don't have one in previous versions.

Reply:
could we put assert somewhere that StateCID is not empty for the new reports while building it?
or we trust what IPFS provider returns us

Reply:
We trust them. Otherwise, we would have a bigger problem.